trunk/src/main/srs_main_ingest_hls.cpp

```diff
@@ -902,25 +902,29 @@ int SrsIngestSrsOutput::parse_message_queue()
             nb_videos--;
         }
         queue.erase(it);
-        
         // parse the stream.
         SrsStream avs;
         if ((ret = avs.initialize(msg->payload->bytes(), msg->payload->length())) != ERROR_SUCCESS) {
             srs_error("mpegts: initialize av stream failed. ret=%d", ret);
+            srs_freep(msg);
             return ret;
         }
         
         // publish audio or video.
         if (msg->channel->stream == SrsTsStreamVideoH264) {
             if ((ret = on_ts_video(msg, &avs)) != ERROR_SUCCESS) {
+                srs_freep(msg);
                 return ret;
             }
         }
         if (msg->channel->stream == SrsTsStreamAudioAAC) {
             if ((ret = on_ts_audio(msg, &avs)) != ERROR_SUCCESS) {
+                srs_freep(msg);
                 return ret;
             }
         }
+        //release detached ts msg
+        srs_freep(msg);
```

Comment:
We could use auto free to free the message.

Reply:
https://github.com/ossrs/srs/commit/6d7c7418a8a21fdfa41a0c023a1bae29ce08b95b#diff-b1f38a5945651bb581b0f2025aca2ef5R901